apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts

```diff
@@ -2,79 +2,80 @@ import { createSelector } from '@reduxjs/toolkit'
 import { DateTime } from 'luxon'
 import { EventTypes } from '@globalfishingwatch/api-types'
 import { t } from 'features/i18n/i18n'
-import { selectFilteredEvents } from 'features/vessels/activity/vessels-activity.selectors'
+import {
+  RenderedEvent,
+  selectFilteredEvents,
+} from 'features/vessels/activity/vessels-activity.selectors'
 import { PortVisitSubEvent } from 'types/activity'
 
-export const selectFilteredEventsWithMainPortVisit = createSelector(
-  [selectFilteredEvents],
-  (events) =>
-    events.reduce((previous, event) => {
-      const isPortExit = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Exit
-      const isPortEntry =
-        event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Entry
+export const getEventsWithMainPortVisit = (events): RenderedEvent[] =>
+  events.reduce((previous, event) => {
+    const isPortExit = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Exit
+    const isPortEntry = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Entry
+
+    if (isPortExit) {
+      // If it's a port exit then build the single port visit event from it
+      const id = event.id.split('-').shift()
+      const portEntry =
+        (isPortEntry && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Entry}`)
```

Comment:
can an event be portExit and portEntry at the same time? if not I think this `(isPortEntry && event)` is not needed
also, could be possible to have portExit without portEntry ?

Reply:
Good point! Based on that I noticed and scenario (having only port visit entry event) where we would not have shown the main port visit so I made some adjustments to fix this problem.